### DISCUSSION TITLE
is it time to retire "dative bonds"?

### DISCUSSION THREAD

**Discussion by nbehrnd** [rdkit/rdkit#6995] [Ideas]
With [Organometallics with dative bonds](http://rdkit.org/docs/Cookbook.html#organometallics-with-dative-bonds), the cookbook includes an example how to process coordination compounds.  The more I read about it, the more I think this type of bonds is a concept of the past and the use of single bonds instead might be better.

I was happy (and still I am) RDKit's cookbook provides the example; with minor adjustments only, it is used to visualize members of a template library for Avogadro (see [here](https://github.com/OpenChemistry/fragments/pull/14)).  In the course of a subsequent discussion in the forum of openbabel ([link](https://github.com/openbabel/openbabel/issues/2656)) I however became aware of two points which may merit a revision:

- In his compilation of rules of structure representations,[1] Brecher labels the use of arrows between the ligand and ion to be complexed as not acceptable (rule GR 1.7.1, page 292).  If RDKit would elect the adoption of this rule, the visual representation would require a change.  The visual result were indiscernible from the one of a single bond (i.e., no arrow).
- IUPAC' online version of the Gold Book includes an entry about [coordination](https://goldbook.iupac.org/terms/view/C01329).  At present, I understand its current revision (February 24, 2014) in lines of «dative bonds represent a concept of the past which should be not be used anymore» because of the following statement:

  > «The synonym 'dative bond' is obsolete. (The origin of the bonding electrons has by itself no bearing on the character of the bond formed. Thus, the formation of methyl chloride from a methyl cation and a chloride ion involves coordination; the resultant bond obviously differs in no way from the C–Cl bond in methyl chloride formed by any other path, e.g. by colligation of a methyl radical and a chlorine atom.)»

Do you equally think the example in RDKit's cookbook merits an update according to the points above?

[1] Graphical representation standards for chemical structure diagrams (IUPAC Recommendations 2008), Pure Appl. Chem., Vol. 80, No. 2, pp. 277–410, 2008. https://doi.org/10.1351/pac200880020277 published open access

**Comment by greglandrum** [comment]
This is an interesting idea, thanks for starting the discussion. Some quick thoughts

Completely getting rid of dative bonds is not an option because, despite the IUPAC recommendations, some software requires them. And they are often (but not always) a useful computationally.

We could think about providing a representation in the RDKit where all dative bonds are converted to single bonds. This would require some thinking about how to handle the electron counts on atoms which are bound to the metal (and would almost certainly require us to stop perceiving implicit Hs on atoms bound to a metal), but it could still be useful

We definitely could relatively easily support a drawing mode where dative bonds are just drawn as single bonds. This would get us closer to complying to the IUPAC drawing recommendations. This could be accompanied by a function which actually converts dative bonds to single bonds so that we could save "complaint" mol files. These might already be useful steps.

**Reply by DavidACosgrove** [reply]
Adding a drawing option DativesAsSingle or such would be very easy.  I can put it on my list.

**Reply by schatzsc** [reply]
I think the drawing isn't even the most serious issue here - much more important is the internal data model and it's use in calculation of implicit hydrogens and "standard valence".

For the graphical output (and only the output), such an option of drawing a straight line (= single bond) instead of "coordinative" and "dative" bonds is possibly not that difficult to realize in terms of software development.

What's a much more serious issue is that depending on the type of metal center, the bonds that connect it with the ligand atoms, the "standard valences" of all atoms involved, and the representation of hydrogens can lead to software "hallucinating" additional hydrogen atoms which do not belong there, leading to incorrect sum formulas (and thus, molecular mass).

**Reply by kienerj** [reply]
The suggested drawing mode and mol file export makes a lot of sense to me.

**Reply by eloyfelix** [reply]
We started exploring the use of V3000 now that InChI has a clear roadmap for enhanced stereochemistry and inorganic/organometallics.
The option of using dative bonds to store consistent molfiles with the drawing option to convert the dative bonds to single ones makes lots of sense to me as well

**Reply by schatzsc** [reply]
As far as I know, the **InChI** team is currently busy with putting the finishing touches on the **v1.07 release** for which a **beta4 version** is available [now](https://github.com/IUPAC-InChI/InChI)

The inorganic/organometallics stereochemistry working group is supposed to meet again later in April or May to discuss and finalize some things but no date fixed yet for the meeting.

**Reply by nbehrnd** [reply]
In similar tune: to ease the construction of small complexes, [Avogadro](https://two.avogadro.cc/) provides a small set predefined ligands.  Based on the discussion here, the visual style of RDKit generated previews of mono- and polydentate ligands was edited in January 2024.  What previously was depicted as a pointy arrow, now is substituted by a plain single bond, e.g.

![example](https://github.com/rdkit/rdkit/assets/19709084/66c2ead1-70b6-4e55-8a99-a62ea0a9d636)

([link to commit](https://github.com/OpenChemistry/fragments/commit/4a2e64793a0b0997e2f77c9028980226d8c26c30))

**Reply by schatzsc** [reply]
An inorganic chemist's dream come true (-:=

But how is it internally represented in the molfile - could you post an example?!?

**Reply by eloyfelix** [reply]
they seemed to have progressed quite a lot on it. Last Tuesday there was a "Progress and plans for InChI" update and things looked quite promising. 

edit: The roadmap image doesn't seem avaiable anymore from their site so removing it from here.

**Reply by schatzsc** [reply]
Where is this roadmap from - seems like you know even more than me?!?

**Reply by eloyfelix** [reply]
edit: roadmap image not avaiable anymore from their site so I dropped it from here as well.

**Reply by nbehrnd** [reply]
@schatzsc No .explicit .mol file is used to generate the previews.  Per group of denticity, there is an annotated list of SMILES in a file `groups.smi`, for instance with an entry reading

```
[CH3]C([O][*]1)=CC(=[O]1)[CH3] a acac-acetylacetonate
```

processed by RDKit by a dedicated script [depict_ligands.py](https://github.com/OpenChemistry/fragments/blob/main/scripts/depict_ligands.py).  The SMILES is the result drawing the ligand in a complex wit Zn on Marvin's [test page](https://marvinjs-demo.chemaxon.com/latest/demo.html); export as SMILES (floppy disk) read by Avogadro.  There, export as .cjson file where the only entry about atomic number `30` is substituted by `0`.  Read again, export Avogadro's SMILES for `groups.smi` the equally storing a marker (`a`, or `m` for automatic/script based, or manual preparation of the preview) and a name.

About 40+ of 54 ligands get their preview this way; difficult cases like flexible EDTA and cyclam, and every haptic ligand (an other 16) are manually prepared with ChemDraw, ChemDoodle, etc.

But I notice my doc string to function `reset_dative_bonds` requires an edit.

**Comment by schatzsc** [comment]
What would be even more beneficial, in my opinion, would be **a "don't touch" option** individually adjustable each atom in the sense that atoms labelled this way will not be subject to any "impicit hydrogen" and "standard valence" routines.

**Comment by schatzsc** [comment]
> We could think about providing a representation in the RDKit where all dative bonds are converted to single bonds. This would require some thinking about how to handle the electron counts on atoms which are bound to the metal (and would almost certainly require us to stop perceiving implicit Hs on atoms bound to a metal), but it could still be useful

In our [TUCAN](https://github.com/TUCAN-nest/TUCAN) demonstrator, we have called this approach **"simple bonds"** in the sense that such bonds only indicate a bonding interaction but do not carry any implications in terms of electron counting.

Also, Brecher in rule _GR-1.7.1 Coordination bonds to single atoms_ mentions that

"Any hydrogen atoms bonded to the atoms at either end of such a coordination bond must be shown clearly, even if that produces a diagram where some atoms appear to have non-standard valences, such as a nitrogen atom with four attached bonds."
